image_generation/lcm_dreamshaper_v7/cpp/src/main.cpp

```diff
@@ -0,0 +1,312 @@
+        ppp.output().model().set_layout("NCHW");
+        ppp.output().tensor().set_layout("NHWC");
+        models.vae_decoder = core.compile_model(vae_decoder_model = ppp.build(), device);
+    }
+
+    // Tokenizer
+    {
+        models.tokenizer = core.compile_model(model_path + "/tokenizer/openvino_tokenizer.xml", device);
+    }
+
+    return models;
+}
+
+ov::Tensor text_encoder(StableDiffusionModels models, std::string& pos_prompt) {
+    const size_t MAX_LENGTH = 77; // 'model_max_length' from 'tokenizer_config.json'
+    const size_t HIDDEN_SIZE = static_cast<size_t>(models.text_encoder.output(0).get_partial_shape()[2].get_length());
+    const int32_t EOS_TOKEN_ID = 49407, PAD_TOKEN_ID = EOS_TOKEN_ID;
+    const ov::Shape input_ids_shape({1, MAX_LENGTH});
+
+    ov::InferRequest tokenizer_req = models.tokenizer.create_infer_request();
+    ov::InferRequest text_encoder_req = models.text_encoder.create_infer_request();
+
+    auto compute_text_embeddings = [&] (std::string& prompt, ov::Tensor encoder_output_tensor) {
+        ov::Tensor input_ids(ov::element::i32, input_ids_shape);
+        std::fill_n(input_ids.data<int32_t>(), input_ids.get_size(), PAD_TOKEN_ID);
+
+        // tokenization
+        tokenizer_req.set_input_tensor(ov::Tensor{ov::element::string, {1}, &prompt});
+        tokenizer_req.infer();
+        ov::Tensor input_ids_token = tokenizer_req.get_tensor("input_ids");
+        std::copy_n(input_ids_token.data<std::int32_t>(), input_ids_token.get_size(), input_ids.data<int32_t>());
+
+        // text embeddings
+        text_encoder_req.set_tensor("input_ids", input_ids);
+        text_encoder_req.set_output_tensor(0, encoder_output_tensor);
+        text_encoder_req.infer();
+    };
+
+    ov::Tensor text_embeddings(ov::element::f32, {1, MAX_LENGTH, HIDDEN_SIZE});
+
+    compute_text_embeddings(pos_prompt, ov::Tensor(text_embeddings, {0, 0, 0}, {1, MAX_LENGTH, HIDDEN_SIZE}));
+
+    return text_embeddings;
+}
+
+ov::Tensor get_w_embedding(float guidance_scale = 7.5, uint32_t embedding_dim = 512) {
+    float w = guidance_scale * 1000;
+    uint32_t half_dim = embedding_dim / 2;
+
+    float emb = log(10000) / (half_dim - 1);
+
+    std::vector<float> emb_vec(half_dim);
+    std::iota(emb_vec.begin(), emb_vec.end(), 0);
+
+    std::transform(emb_vec.begin(), emb_vec.end(), emb_vec.begin(),
+                    std::bind(std::multiplies<float>(), std::placeholders::_1, -emb));
+
+    std::transform(emb_vec.begin(), emb_vec.end(), emb_vec.begin(), [](float x){return std::exp(x);});
+
+    std::transform(emb_vec.begin(), emb_vec.end(), emb_vec.begin(),
+                    std::bind(std::multiplies<float>(), std::placeholders::_1, w));
+
+    std::vector<float> res_vec(half_dim), emb_cos(half_dim);
+    std::transform(emb_vec.begin(), emb_vec.end(), res_vec.begin(), [](float x){return std::sin(x);});
+    std::transform(emb_vec.begin(), emb_vec.end(), emb_cos.begin(), [](float x){return std::cos(x);});
```

Comment:
looks like we can remove this multiple `std::transform` calls with simple `for` loop, which performs several operations at once